### PR TITLE
Support reuse of already existing resource

### PR DESCRIPTION
**Pull request by myakove**
For debug, support reuse an existing resource and skip the creation

### CODE REVIEW

**Comment by myakove** [comment]
!verified

File: ocp_resources/resource.py [hunk 1]
```diff
@@ -446,16 +450,53 @@ def to_dict(self):
 
     def __enter__(self):
         signal(SIGINT, self._sigint_handler)
-        return self.deploy()
+        """
+        For debug, export REUSE_IF_RESOURCE_EXISTS to skip resource create.
```
Comment (sarahbx) [inline ocp_resources/resource.py:454]: could this be an `__init__` variable instead of environment?

File: ocp_resources/resource.py [hunk 2]
```diff
@@ -446,16 +450,53 @@ def to_dict(self):
...
+        Examples:
+            To skip creation of all resources by kind:
+                export REUSE_IF_RESOURCE_EXISTS="{Pod: {}}"
+
+            To skip creation of resource by name (on all namespaces):
```
Comment (rnetser) [inline ocp_resources/resource.py:459]:
```suggestion
            To skip creation of resource by name (on all namespaces or non-namespaced resources):
```

File: ocp_resources/resource.py [hunk 3]
```diff
@@ -446,16 +450,53 @@ def to_dict(self):
...
+        return _resource or self.deploy()
 
     def __exit__(self, exception_type, exception_value, traceback):
-        # For debug, export kind (class name) + NOTEARDOWN to skip resource teardown.
-        # For example:export VirtualMachineInstanceNOTEARDOWN=True
-        skip_teardown_env = f"{self.kind}NOTEARDOWN"
-        no_teardown_from_environment = os.environ.get(skip_teardown_env)
-        if no_teardown_from_environment:
+        """
+        For debug, export SKIP_RESOURCE_TEARDOWN to skip resource create.
```
Comment (rnetser) [inline ocp_resources/resource.py:479]:
```suggestion
        For debug, export SKIP_RESOURCE_TEARDOWN to skip resource teardown.
```

relevant in the comment below as well

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/resource.py (hunk 1 above)
Comment (myakove) [reply ocp_resources/resource.py:454]: No, the point of this is to let the user control it without any code change.

**Comment by myakove** [comment]
!verified

**Comment by myakove** [comment]
!verified

**Comment by myakove** [comment]
!verified

**Comment by myakove** [comment]
!verified

**Review by sarahbx — APPROVED** [review]

**Review by rnetser — APPROVED** [review]

**Pull request merged**